### ISSUE TITLE
`testexpect` tests started to fail

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#5309]
The `testexpect` test in CI has started to fail recently, and I am not sure why. Examples:
- https://github.com/gap-system/gap/actions/runs/3867639765/jobs/6592560167
- https://github.com/gap-system/gap/actions/runs/3873539074/jobs/6603676034
- https://github.com/gap-system/gap/actions/runs/3854722893/jobs/6600949814

That said, it worked fine in the last daily build... 

Right now I am suspecting that perhaps something changed in the CI container. I've only just now started to look.

CC @ThomasBreuer (who pointed this out to me this morning) and @ChrisJefferson (who might have more ideas)

**Comment by fingolfin** [comment]
Another potential cause might of course be timing issues, to which `expect` is always a bit sensitive. But in this case, it looks as if `ctrl-U` (to delete the content of the input line) is not executed, or least not before `enter` is processed, and I have a hard time seeing how that could be caused by delays... still, it might be worth a try to insert "wait" statements... Though ideally one would first be able to reproduce the issue, which I am currently not: running this command always work (but maybe my computer is "too fast" to trigger the issue)
```
TEST_SUITES=testexpect dev/ci.sh
```

**Comment by fingolfin** [comment]
The failures went away, kinda, but now we got one again: in [this job](https://github.com/gap-system/gap/actions/runs/4063767814/jobs/6996390017) from PR #3450:
```
2023-02-01T11:16:58.5740338Z Running test suite testexpect
2023-02-01T11:16:58.5741111Z + expect -c 'spawn ./gap --quitonbreak -A -b  --cover coverage/testexpect.coverage' /home/runner/work/gap/gap/dev/gaptest.expect
2023-02-01T11:16:58.7295645Z spawn ./gap --quitonbreak -A -b --cover coverage/testexpect.coverage
2023-02-01T11:17:01.2043439Z [1m[34mgap> [0m[31mvar := rec(member := true, data1 := 1, data2 := 2);;
2023-02-01T11:17:01.2256094Z [0m[1m[34mgap> [0m[31mvar
2023-02-01T11:17:01.2256594Z var  
2023-02-01T11:17:01.2258611Z gap> var;
2023-02-01T11:17:01.2263663Z [0mrec( data1 := 1, data2 := 2, member := true )
2023-02-01T11:17:01.3293524Z [1m[34mgap> [0m[31mvar.data
2023-02-01T11:17:01.3294530Z data1  data2  
2023-02-01T11:17:01.4299991Z gap> var.data1;
2023-02-01T11:17:01.4300824Z [0m1
2023-02-01T11:17:01.4326928Z [1m[34mgap> [0m[31mReturnTrue(var.data
2023-02-01T11:17:01.4327165Z data1  data2  
2023-02-01T11:17:01.4331071Z gap> ReturnTrue(var.data1);
2023-02-01T11:17:01.4333558Z [0mtrue
2023-02-01T11:17:01.4334078Z [1m[34mgap> [0m[31mg:=SmallGroup(6,1);;
2023-02-01T11:17:01.4339331Z g:=SmallGroup(6,1);;
2023-02-01T11:17:01.4385880Z [0m[1m[34mgap> [0m[31mg.	1;;
2023-02-01T11:17:01.4490213Z g.1;;
2023-02-01T11:17:01.4508065Z [0m[1m[34mgap> [0m[31mg!.Size;;
2023-02-01T11:17:01.4511117Z [0m[1m[34mgap> [0m[31mg.x.	    
2023-02-01T11:17:01.4632756Z [0m[1m[34mgap> [0m[31mg!.x.
2023-02-01T11:17:01.4635414Z [0mError, Record Element: '<rec>.x' must have an assigned value
2023-02-01T11:17:01.4678209Z  EOF 
2023-02-01T11:17:01.5129377Z ##[error]Process completed with exit code 1.
```

**Comment by ThomasBreuer** [comment]
The problem occurs now in a test run for PR #5460.

```
Thu, 08 Jun 2023 12:21:20 GMT Running test suite testexpect
Thu, 08 Jun 2023 12:21:20 GMT + cd /home/runner/work/gap/gap
Thu, 08 Jun 2023 12:21:20 GMT + echo 'Running test suite testexpect'
Thu, 08 Jun 2023 12:21:20 GMT + case $TEST_SUITE in
Thu, 08 Jun 2023 12:21:20 GMT + INPUTRC=/tmp/inputrc
Thu, 08 Jun 2023 12:21:20 GMT + expect -c 'spawn ./gap --quitonbreak -A -b  --cover coverage/testexpect.coverage' /home/runner/work/gap/gap/dev/gaptest.expect
Thu, 08 Jun 2023 12:21:20 GMT spawn ./gap --quitonbreak -A -b --cover coverage/testexpect.coverage
Thu, 08 Jun 2023 12:21:23 GMT gap> var := rec(member := true, data1 := 1, data2 := 2);;
Thu, 08 Jun 2023 12:21:23 GMT gap> var
Thu, 08 Jun 2023 12:21:23 GMT var  
Thu, 08 Jun 2023 12:21:23 GMT gap> var;
Thu, 08 Jun 2023 12:21:23 GMT rec( data1 := 1, data2 := 2, member := true )
Thu, 08 Jun 2023 12:21:23 GMT gap> var.data
Thu, 08 Jun 2023 12:21:23 GMT data1  data2  
Thu, 08 Jun 2023 12:21:23 GMT gap> var.data1;
Thu, 08 Jun 2023 12:21:23 GMT 1
Thu, 08 Jun 2023 12:21:23 GMT gap> ReturnTrue(var.data
Thu, 08 Jun 2023 12:21:23 GMT data1  data2  
Thu, 08 Jun 2023 12:21:23 GMT gap> ReturnTrue(var.data1);
Thu, 08 Jun 2023 12:21:23 GMT true
Thu, 08 Jun 2023 12:21:23 GMT gap> g:=SmallGroup(6,1);;
Thu, 08 Jun 2023 12:21:23 GMT g:=SmallGroup(6,1);;
Thu, 08 Jun 2023 12:21:23 GMT gap> g.1;;
Thu, 08 Jun 2023 12:21:23 GMT gap> g!.Size;;
Thu, 08 Jun 2023 12:21:23 GMT gap> g.x.
Thu, 08 Jun 2023 12:21:23 GMT Error, illegal access to record component `obj.x'
Thu, 08 Jun 2023 12:21:23 GMT of the object <obj>. (Objects by default do not have record components.
Thu, 08 Jun 2023 12:21:23 GMT The error might be a relic from translated GAP3 code.) at /home/runner/work/gap/gap/lib/record.gi:160 called from
Thu, 08 Jun 2023 12:21:23 GMT  EOF 
Thu, 08 Jun 2023 12:21:23 GMT Error: Process completed with exit code 1.
```

**Comment by fingolfin** [comment]
This hasn't occurred recently and IIRC it may have been related to (and fixed by) #5504
